src/inverse-kinematics/tests/ConvexHullHelpersUnitTest.cpp

```diff
@@ -97,19 +97,37 @@ void testConvexHullProjectionWithGravity()
     // Note: positive margin means inside, negative outside
     ASSERT_IS_TRUE(projectionConstraint.computeMargin(projectionConstraint.project(testPoint)) < 0);
 
-    // If the direction along with we are projecting is the normal of the plane, the projection
+    // If the direction along witch we are projecting is the normal of the plane, the projection
     // should match the one done without direction
-    iDynTree::Direction planeNormal(0.0, 0.0, -1.0);
+
+    //iDynTree::Direction planeNormal(0.0, 0.0, -1.0);
     //ASSERT_EQUAL_VECTOR(projectionConstraint.project(testPoint), projectionConstraint.projectAlongDirection(testPoint, planeNormal));
+    iDynTree::Vector3 planeNormalVector;
+    planeNormalVector.setVal(0, 0.0);
+    planeNormalVector.setVal(1, 0.0);
+    planeNormalVector.setVal(2, -1.0);
+
+    projectionConstraint.setProjectionAlongDirection(planeNormalVector, iDynTree::Position::Zero());
+
+    ASSERT_EQUAL_VECTOR(projectionConstraint.project(testPoint), projectionConstraint.projectAlongDirection(testPoint));
+
+    //ASSERT_IS_TRUE(false);
+
+    // If the direction of the projection is "skewed" to the left, the point should instead be inside the convex hull
+    //iDynTree::Direction skewedDirection(-1.0, 0.0, -1.0);
+    //ASSERT_IS_TRUE(projectionConstraint.computeMargin(projectionConstraint.projectAlongDirection(testPoint, skewedDirection)) < 0);
+    iDynTree::Vector3 skewedDirection;
+    skewedDirection.setVal(0, -1.0);
+    skewedDirection.setVal(1, 0.0);
+    skewedDirection.setVal(2, 1.0);
```

Comment:
I think a `projectionConstraint.setProjectionAlongDirection(skewedDirection, iDynTree::Position::Zero())` is missing here (and this is one of the reason for which a single call `projectAlongDirection(point,direction)` could be more handy.

Reply:
Yes, I forgot it. Thanks!

Reply:
So would it be better to store the `direction` vector inside `InverseKinematics`?

But considering the possible uses of the method `projectAlongDirection`, I think it's more logic to keep it associated with the class `ConvexHullProjectionConstraint`. What do you think? @traversaro

Reply:
I think the API would be easier to use by `projectAlongDirection(point,direction)`, and for any use in `InverseKinematics` it can always be called by passing the projection direction saved in `InverseKinematics` . In this case we add an attribute in `InverseKinematics`, but eventually we will need to refactor that logic in its own class, so I think it is reasonable to do so.